### PR TITLE
Modify the UX flow for deploying contract #187

### PR DESCRIPTION
**Pull request by bradysheridan**
Done
- [x] Implement first design iteration
- [x] Set up transitions between steps
- [x] Bind transitions to events
- [x] Integrate into `Quick`, `Guided`, and `Simple` deployment flows
- [x] Write test code for all new React components

##### Screenshots (Optional) 
Before:

screen shot 2018-06-21 at 2 43 09 pm

screen shot 2018-06-21 at 2 44 21 pm

After:

screen shot 2018-06-21 at 2 41 37 pm

##### Refers/Fixes
Fixes: #187

### CODE REVIEW

**Comment by bradysheridan** [comment]
@pelsasser @eswarasai 
Here's the WIP PR! Could you guys provide any guidance on binding the steps in this flow to actual deployment events?

**Comment by eswarasai** [comment]
@bradysheridan -- Sure. So in the [`src/actions/deploy`](https://github.com/MARKETProtocol/dApp/blob/develop/src/actions/deploy.js#L10), we call the deploy contract methods and handle the contract deployment accordingly. I'd recommend you to add few variables to [`state`](https://github.com/MARKETProtocol/dApp/blob/develop/src/reducers/deploy.js#L2) here and update these variables on success of each step via `actions` and subscribe to them in the `Deployment` Component. Let me know if you need any further assistance w.r.t. technical implementation. Thanks!

**Comment by bradysheridan** [comment]
@eswarasai Thanks, that's super helpful!

**Comment by bradysheridan** [comment]
@eswarasai 
This has been more or less fully implemented, I'm just having trouble integrating it into the `Quick` and `Guided` deployment flows as they both require a token address input--what should I enter in this field?

**Comment by eswarasai** [comment]
@bradysheridan -- Can you guide me on how you're integrating into `Quick` and `Guided` deployment and also where is the token address input prop needed?

PS: I haven't reviewed the PR yet. Let me take a look and get back to you in case I can get them done for you.

**Comment by bradysheridan** [comment]
@eswarasai Ah so I'm not talking about a prop, I'm just referring to the token address input field in the contract deployment flow. This one:

screen shot 2018-06-23 at 10 40 32 am

I'm not sure what to enter here and so I'm not able to progress through these flows.

**Comment by bradysheridan** [comment]
@eswarasai @pelsasser 
Should be all set save for test code, which I'll wait to write until after you give me feedback, if you have any!

File: src/less/ContractDeploymentFlow.less [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+.hide-on-mobile {
+    @media screen and (max-width: 576px) {
+        display: none !important;
+    }
+}
+
+.ant-modal {
+    width: 90% !important;
```
Comment (eswarasai) [inline src/less/ContractDeploymentFlow.less]: @bradysheridan -- This is effecting the `modal` widths at other places. For example on the landing screen, welcome modal.

**Comment by eswarasai** [comment]
@bradysheridan -- Overall the UI/UX of the steps for the contract deployment flow looks great. There are minor issues in regards with the success and failure of the contract deployment flow. 

For instance, if a contract deployment fails, there's no Go back/Retry button for the User to deploy the contract again and also I don't think the state is being reset upon success/failure.

Is it possible to avoid the deployment of contract steps in a modal for Quick deployment flow? It's looking a bit weird in the modal compared to the Simplified and Guided Deployment flow. Maybe, you can hide the Quick deployment form while deploying the contract based on a prop and provide a back button to go back from deployment steps screen to the Quick deployment form.

File: src/less/ContractDeploymentFlow.less (hunk 1 above)
Comment (bradysheridan) [reply src/less/ContractDeploymentFlow.less]: Gotcha, will remedy!

**Comment by bradysheridan** [comment]
@eswarasai Sorry for the delay, I've been in airplane limbo for the last day or so. I'll implement a `Go Back` button, fix the state update issue, move the flow from a modal to a regular screen, and fix the style overrides that are affecting other parts of the app.

**Comment by eswarasai** [comment]
@bradysheridan -- Sure. No problem. Let us know once it's ready for review. Will share the feedback, if any.

**Comment by eswarasai** [comment]
@bradysheridan -- Just checking in to see how's this going so far?

**Comment by bradysheridan** [comment]
@eswarasai Hey, really sorry about the snail-paced turn around on this. I just settled into a new city and started a language program here in France, but I'll have more free time now. I'm posted at a Starbucks right now finishing this up!

**Comment by eswarasai** [comment]
@bradysheridan -- No worries. Let me know if you need anything from my end in order to knock this out. Thanks :)

**Comment by bradysheridan** [comment]
@eswarasai I've done everything I mentioned here:

> I'll implement a Go Back button, fix the state update issue, move the flow from a modal to a regular screen, and fix the style overrides that are affecting other parts of the app.

I'm going to write the test code over the course of the next couple of nights and will ping you when it's done.

**Comment by eswarasai** [comment]
@bradysheridan -- Thanks for the update. Will review the changes and let you know in case there's anything else needed :)

**Comment by pelsasser** [comment]
@bradysheridan - excited to get this merged when you have some time to finish it off! Please let us know if we can help in any way.

**Comment by eswarasai** [comment]
@bradysheridan -- How's this going? We've recently merged lot of changes to `upstream:develop`. I'd recommend rebasing your branch with that of `upstream:develop` and resolve any conflicts. Thanks!

**Comment by eswarasai** [comment]
@bradysheridan -- Just checking in to see the progress so far and also what are the pending things in order to wrap up this PR. Thanks!

**Comment by bradysheridan** [comment]
@eswarasai If the coverage suits you, it's good to go.

**Comment by eswarasai** [comment]
@bradysheridan -- Thanks. I'll review and try to get it merged ASAP. Yeah, we can tackle the overall coverage in a separate issue.

**Comment by eswarasai** [comment]
@bradysheridan -- PR is looking good by me to merge. Can you please take care of the conflicts? Also, note that there has been some changes to the way the `Loader` component is being used. Let me know if you need any help with resolving the conflicts, I can take care of them for you. Thanks!

**Comment by eswarasai** [comment]
@bradysheridan -- I've resolved the conflicts and will be merging this PR now. We'll handle any required changes internally. Thanks a lot for handling the end-to-end functionality. Really appreciate it.

If you can submit work on Gitcoin, @pelsasser can pay you out :)

**Comment by bradysheridan** [comment]
@eswarasai Great! Thank you! Just submitted work on Gitcoin.

**Comment by bradysheridan** [comment]
@eswarasai And I apologize again for the terribly slow turnaround. I'll be sure to time my bounties better in the future.

**Comment by pelsasser** [comment]
@bradysheridan - thanks again for getting this done, just used it on rinkeby and very much like it!

**Pull request merged**